src/main/java/billgates/view/gui/AddEntryDialog.java

```diff
@@ -0,0 +1,97 @@
+package billgates.view.gui;
+
+import com.github.lgooddatepicker.components.DateTimePicker;
+
+import javax.swing.*;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+public class AddEntryDialog extends JDialog {
+
+    private int result;
+    private ZonedDateTime date;
+    private static double value;
+    private String currency;
+    private String description;
+    private String from;
+    private String to;
+    private String location;
+    private final MainFrame mainFrame;
+
+    public AddEntryDialog(MainFrame owner) {
+        super(owner);
+        this.mainFrame = owner;
+        // Create a datetime picker.
+        DateTimePicker dateTimePicker1 = new DateTimePicker();
+        add(dateTimePicker1);
+
+        JTextField valueField = new JTextField(20);
+        JTextField currencyField = new JTextField(20);
+        JTextField descriptionField = new JTextField(20);
+        JTextField fromField = new JTextField(20);
+        JTextField toField = new JTextField(20);
+        JTextField locationField = new JTextField(20);
+
+        JPanel myPanel = new JPanel();
```

Comment:
Small naming convention, maybe we don't want to call this object `myPanel`, we can have the name being something more useful, something like `insertEntryDialog`